public/Test-DbaDbCompatibility.ps1

```diff
@@ -77,7 +77,8 @@ function Test-DbaDbCompatibility {
                 Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
             }
 
-            $serverLevel = [Microsoft.SqlServer.Management.Smo.CompatibilityLevel]$server.Databases['master'].CompatibilityLevel
+            $serverVersion = $server.VersionMajor
+            $serverLevel = [Microsoft.SqlServer.Management.Smo.CompatibilityLevel]"Version$($serverVersion)0"
```

Comment:
This is what was being done in Set command as well and it was removed because it is unsafe method.

Reply:
So we need two internal commands the have a rubust way to translate from MajorVersion to CompatibilityLevel and vice versa. I agree with that. We could then use this in many places...
The question is: Is there an "arithmetic rule" or do we need a switch with hard coded values?

Reply:
https://stackoverflow.com/questions/59618625/is-there-a-way-to-query-for-the-supported-compatibility-level-of-a-sql-server suggested using tempdb. That could be a way, will test...

Reply:
I'm still confused on why we think it is necessary to translate between MajorVersion and Compatibility level. If there was a reason for that then SMO should provide it not us because we are assuming we know the mapping. If we are going to keep a mapping then revert the changes done in Set command and replace that use everywhere.

Reply:
For Test-DbaDbCompatibility (Compares Database Compatibility level to Server Compatibility) we need to know the highest possible Compatibility level on that server. So we have the server SMO and need that highest possible Compatibility level. Where do we get that?